openerp/service/server.py

```diff
@@ -232,7 +232,12 @@ def close_socket(self, sock):
         sock.close()
 
 class ThreadedServer(CommonServer):
+    initial_threads = []
+    
     def __init__(self, app):
+        for t in threading.enumerate():
+            self.initial_threads.append(t)
```

Comment:
You're missing my point. In their normal course of operation, some modules will spawn new threads to perform tasks asynchronously, the shutdown procedure waits for them so they can finish whatever they're doing instead of being interrupted. 

Your patch breaks that, and it functionally removes all waiting since *you're only whitelisting the main thread which is specifically not being waited on*.

Since PyDev's injected thread is never shut down and should not block terminating the software, it should be daemonified, that's the entire point of that flag.